packages/framework-provider-local/src/services/subscription-registry.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { SubscriptionEnvelope } from '@boostercloud/framework-types'
+import * as DataStore from 'nedb'
+import { subscriptionDatabase } from '../paths'
+
+export class SubscriptionRegistry {
+  public readonly subscriptions: DataStore<SubscriptionEnvelope> = new DataStore(subscriptionDatabase)
+
+  constructor() {
+    this.subscriptions.loadDatabase()
+  }
+
+  getCursor(query: object, createdAt = 1, projections?: unknown) {
+    return this.subscriptions.find(query, projections).sort({ createdAt: createdAt })
+  }
+
+  public async query(query: object, createdAt = 1, limit?: number, projections?: unknown): Promise<unknown> {
+    const cursor = this.getCursor(query, createdAt, projections)
+    if (limit) {
+      cursor.limit(Number(limit))
+    }
+    const queryPromise = new Promise((resolve, reject) => {
+      cursor.exec((err, docs) => {
+        if (err) reject(err)
+        else resolve(docs)
+      })
+    })
+
+    return await queryPromise
+  }
+
+  public async store(subscriptionEnvelope: SubscriptionEnvelope): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.subscriptions.insert(subscriptionEnvelope, (err) => {
+        err ? reject(err) : resolve()
+      })
+    })
+  }
+
+  public async delete(query: unknown): Promise<number> {
+    const deletePromise = new Promise((resolve, reject) =>
+      this.subscriptions.remove(query, { multi: true }, (err, numRemoved: number) => {
+        if (err) reject(err)
+        else resolve(numRemoved)
+      })
+    )
+
+    return (await deletePromise) as number
+  }
+
+  public async deleteAll(): Promise<number> {
+    const deletePromise = new Promise((resolve, reject) =>
+      this.subscriptions.remove({}, { multi: true }, (err, numRemoved: number) => {
+        if (err) reject(err)
+        else resolve(numRemoved)
+      })
+    )
+
+    return (await deletePromise) as number
+  }
+}
```

Comment:
This class is (if I didn't miss something) exactly the same as ConnectionRegistry. We can have only one class (for example called SimpleRegistry) that has the database path as the only parameter. This way, we avoid duplicating all the code.

Reply:
Good point